### DISCUSSION TITLE
CheckBox v2 vs v3 Callback

### DISCUSSION THREAD

**Discussion by danchitnis** [chakra-ui/chakra-ui#9003] [Q&A] [answered]
Hello, I am trying to migrate my code from v2 to v3. The checkbox seems to have undergone a major overhaul. My question is about the callback for v3, which is `CheckedChangeDetails`, and only consists of:

```typescript
interface CheckedChangeDetails {
    checked: boolean;
}
```

In the old version, the callback was `HTMLInput`, and it contained all the event details, including the `name` and `id` of the item that triggered the event. 

I am confused about how I can get the `name` of the checkbox that triggered the event from the callback since it no longer exists in the callback information?

**Comment by segunadebayo** [comment]
Do you have a reproduction or code you can show? I'm happy to help you with that.

**Comment by danchitnis** [comment]
@segunadebayo Thanks! This is the code based on Chakra V2. The checkboxes are generated [here](https://github.com/eelab-dev/EEcircuit/blob/83114a5162bb8ed6e31bb6ac607a5258dbd0fd84/src/displayBox.tsx#L38) with a unique `name` prop. Then the `onChange()` callback is handled [here](https://github.com/eelab-dev/EEcircuit/blob/83114a5162bb8ed6e31bb6ac607a5258dbd0fd84/src/EEcircuit.tsx#L199).

The event handler checks for the `name` prop of the triggered event and, based on the `name` makes a decision on what to change. This is possible because all the props related to the checkbox which triggered the event are encapsulated in the callback input `e.target`.

**Comment by danchitnis** [comment]
I have been looking more into this. It seems [CheckboxGroup](https://www.chakra-ui.com/docs/components/checkbox#group) and [CheckBoxCard](https://www.chakra-ui.com/docs/components/checkbox-card) are more suitable here. But I still don't know how to handle the events coming from a checkbox card or group. Currently, there should be a separate callback function for each check item, which seems very inefficient. Is there a specific reason why the `name` and `id` props are excluded from the `CheckedChangeDetails`?

**Comment by danchitnis** [comment, marked as answer]
I solved this by bundling the name into the callback function:

```typescript
onCheckedChange={(e: CheckboxCheckedChangeDetails) => 
   checkCallBack(checkbox.name, e.checked === true ? true : false)}
```

Then pass is a prop for the components, which contains the checkbox:

```typescript
type Props = {
  ...
  checkCallBack: (name: string, check: boolean) => void;
  ...
};
```

And the callback handler in parent components, which contains the checkbox component:

```typescript
const change = React.useCallback(
    (name: string, check: boolean) => {...}
```
